File: Lista2/Questao1/main.cpp
#include <iostream>

int main()
{
 
    int i, n = 16;
    for (i = 1; i <= n; i=i*2) {
        std::cout << "Hello World !!!\n";
    }
    return 0;
}

//explicação : aqui temos uma iteração que cresce duas vezes de tamanho na leitura. É uma estrutura linear pois ainda vai ler um conjunto de dados passado anteriormente e vai crescer de acordo com a leitura, fazendo desse método um método de complexidade O(n)

File: Lista2/Questao1/main.py
def quadratic_algo(items):
    for item in items:
        for item2 in items:
            print(item, ' ' ,item2)

quadratic_algo([4, 5, 6, 8])

#Explicação : nesse método há duas estruturas lineares, uma dentro da outra, fazendo com que cada iteração do primmeiro conjunto de dados(array), gere uma segunda estrutura linear inteira, assim temos  O(n) x O(n) = O(n²)
